lib/dialects/abstract/query-generator.js

```diff
@@ -1766,6 +1767,11 @@ module.exports = (function() {
         value = (value.$between || value.$notBetween).map(function (item) {
           return self.escape(item);
         }).join(' AND ');
+      } else if (value && ((value.$like && (any = value.$like.any) && (comparator = 'LIKE ANY')) ||
```

Comment:
Not the most readable piece of code in the world but get's the job done i suppose, and not sure i can come up with anything more elegant right now.

Would it be too much hassle to ask you to test all the ANY paths? ($like, $iLike, $notLike, $notILike)